docs: rfc-9 scenes preview server (#92)
decentraland/adr: RFC/RFC-9-hosted-preview-service.md

---
layout: doc
rfc: 9
date: 2022-09-28
title: Scene Preview Server
authors:
- pentreathm
status: DRAFT
---

# Abstract

Today users that don't own a land or are not ready to push their content to production need to use a Heroku application container to be able to preview and share a scene with different stakeholders. This solution requires some extra configuration steps and, on the other hand, it will stop being free according to the [latest Heroku news](https://blog.heroku.com/next-chapter). To ease the life of the content creators, a new scene preview server will be created, leveraging the same steps than when uploading content to the Catalyst network and enabling an easy way to validate and experience the content before it is finally uploaded to the Genesis City.  
 
## Approach

The solution for the scenes preview will be done in stages

> ### Stage 1: Preivew Server BETA 

Reviewer: ```suggestion
### Stage 1: Preview Server BETA 
```

Create a BETA program where the technical feasibility of the project can be evaluated and test the experience from the content creators perspective.  

Scope of the BETA: 
- Open to users owning a Decentraland name 
- Only one scene per Decentraland name would be allowed on the preview 
- Scenes deployment to the preview server will have the same disk space limitations that the Content Servers hosted on the Catalyst network 

> - Besides space and owning a name, no other validations will be made as this is just a preview

Reviewer: ```suggestion
- Besides scene size and owning a name, no other validations will be made as this is just a preview
```

`dcl deploy -t https://example-url-for-dcl-preview-server`  
 
 The output of this command should contain a link that can be shared and used to jump in to the scene preview, e.g. `https://play.decentraland.org/?PREVIEW_SERVER=...&SCENE=...&COMMS=....` 

From the Client and Services side, this will work very similar as with the existing Catalyst nodes. The client will receive several new parameters: **server** from which to resolve the scene, **scene identifier**, and the **comms server URL**, in charge of managing the communications between users visiting the scene preview. On the other hand, the existing Catalyst network will be used to resolve the users digital identity, retrieving their avatars and wearables.  

```mermaid
flowchart TD
    B["Explorer Client"]
    B<-- scene-id -->C(Preview Content Server)
    B<-->D(Preview Comms Server);
    E("Catalyst") -- avatar -->B
```

When the Client loads a scene from the Preview server, the Mini map should not be loaded as the scene is going to be isolated from the Genesis City.

![mini-map](img/rfc-9/minimap.png)

As a result: 
- The learning experience to upload and preview a scene will be minimal 

> - Scenes content can be tested multi-player and isolated from the Genesis City   
> - The preview UX can be tested with a subset of content creators   

Reviewer: ```suggestion
- Scenes content can be tested multi-player and isolated from the Genesis City   
- Content Creators can test the preview UX of their scenes   
```

> - As the access will be limited, all the focus is going to be on the experience and scalability will come on the following stage

Reviewer: ```suggestion
- On this stage, all the focus is going to be on content creator user experience
```

---
layout: doc
rfc: 9
date: 2022-09-28
title: Scene Preview Server
authors:
- pentreathm
status: DRAFT
---

# Abstract

> Today users that don't own a land or are not ready to push their content to production need to use a Heroku application container to be able to preview and share a scene with different stakeholders. This solution requires some extra configuration steps and, on the other hand, it will stop being free according to the [latest Heroku news](https://blog.heroku.com/next-chapter). To ease the life of the content creators, a new scene preview server will be created, leveraging the same steps than when uploading content to the Catalyst network and enabling an easy way to validate and experience the content before it is finally uploaded to the Genesis City.  

Reviewer 2: This section rather an introduction instead of an abstract, the abstract should reflect a tiny short summary of the whole document

Author: updated

 
## Approach

The solution for the scenes preview will be done in stages

> ### Stage 1: Preview Server BETA 

Reviewer 2: I'd ditch the BETA words, we are already talking about stages

Author: removed

Create a BETA program where the technical feasibility of the project can be evaluated and test the experience from the content creators perspective.  

> Scope of the BETA: 

Reviewer 2: Can we add to the scope the configurations of day/night, hiding mini map, going back and forth from the genesis city and use it to host the onboarding tutorial?

Author: added

- Open to users owning a Decentraland name 
- Only one scene per Decentraland name would be allowed on the preview 
- Scenes deployment to the preview server will have the same disk space limitations that the Content Servers hosted on the Catalyst network 
- Besides scene size and owning a name, no other validations will be made as this is just a preview

> - 10 concurrent users allowed to join a scene preview

Reviewer 2: I think it is more complicated to add this limitation than not having it

Author: Removed restriction. We can react to this if needed.
